### PR TITLE
[NEXUS-7333] support ehcache custom configuration

### PR DESCRIPTION
**Pull request by mcculls**
https://issues.sonatype.org/browse/NEXUS-7333
http://bamboo.s/browse/NX3-OSSF322-2 :white_check_mark:

### CODE REVIEW

File: components/nexus-ehcache/src/main/java/org/sonatype/sisu/ehcache/CacheManagerComponentImpl.java [hunk 1]
```diff
@@ -48,8 +50,8 @@
   private CacheManager cacheManager;
 
   @Inject
-  public CacheManagerComponentImpl() throws IOException {
-    this(null);
+  public CacheManagerComponentImpl(final ApplicationDirectories directories) throws IOException {
+    this(directories.getAppDirectory("etc/ehcache.xml"));
```
Comment (jdillon) [inline components/nexus-ehcache/src/main/java/org/sonatype/sisu/ehcache/CacheManagerComponentImpl.java]: odd use of method, safer to use:

    new File(directories.getAppDirectory("etc"), "ehcache.xml")

Comment (mcculls) [inline components/nexus-ehcache/src/main/java/org/sonatype/sisu/ehcache/CacheManagerComponentImpl.java]: presumably this should also be changed: https://github.com/sonatype/nexus-oss/blob/master/components/nexus-elasticsearch/src/main/java/org/sonatype/nexus/elasticsearch/internal/NodeProvider.java#L79 ?

Comment (jdillon) [inline components/nexus-ehcache/src/main/java/org/sonatype/sisu/ehcache/CacheManagerComponentImpl.java]: ya, we could improve this helper to have get*File() variants but I want to wait a bit before adding stuff like that, as I think we need nio helpers too.

But ya, the component as-is was meant to return directories (and if we ever added sanity checks to ensure they are directories if they exist, things would blow up).

**Comment by jdillon** [comment]
+1

**Comment by cstamas** [comment]
+1

**Comment by mcculls** [comment]
Merged to master as 9e1103d

**Pull request closed without merging**